src/test/java/de/tum/cit/aet/artemis/shared/architecture/ArchitectureTest.java

```diff
@@ -384,6 +389,16 @@ private boolean hasMatchingTestClassCallingMethod(JavaClass javaClass, String id
         return false;
     }
 
+    @Test
+    void ensureSpringComponentsAreLazyAnnotated() {
+        ArchRule rule = classes().that().areAnnotatedWith(Controller.class).or().areAnnotatedWith(RestController.class).or().areAnnotatedWith(Repository.class).or()
+                .areAnnotatedWith(Service.class).or().areAnnotatedWith(Component.class).or().areAnnotatedWith(Configuration.class).should().beAnnotatedWith(Lazy.class)
+                .because("All Spring components should be lazy-loaded to improve startup time");
+
+        rule.check(classesExcept(allClasses, ApplicationConfiguration.class, ConditionalMetricsExclusionConfiguration.class, SecurityConfiguration.class, TomcatConfiguration.class,
+                HibernateQueryInterceptor.class, HibernatePropertiesConfig.class));
```

Comment:
How do we know that these are the only configuration classes that we need to exclude?